pkg/operator/controllers/imageconfig/image_controller.go

```diff
@@ -0,0 +1,149 @@
+package imageconfig
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+)
+
+const imageConfigResource = "cluster"
+
+const (
+	CONFIG_NAMESPACE string = "aro.imageconfig"
+	ENABLED          string = CONFIG_NAMESPACE + ".enabled"
```

Comment:
* Should not be exported as we only use it internally
* Should be camelCase

But I tihnk it was based on another controller with flags. Need to update them, but it is out of scope for this PR. Let's merge like this and do a clean up. Probably as part of #1925